Packages/com.unity.inputsystem/InputSystem/Plugins/iOS/GameController.cs

```diff
@@ -0,0 +1,83 @@
+using System.Runtime.InteropServices;
+using ISX.Controls;
+using ISX.LowLevel;
+using ISX.Utilities;
+using UnityEngine;
+
+namespace ISX.iOS
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public unsafe struct GameControllerState : IInputStateTypeInfo
+    {
+        enum Button 
+        {
+            DpadUp,
+            DpadDown,
+            DpadLeft,
+            DpadRight,
+            LeftStick,
+            RightStick,
+            LeftShoulder,
+            RightShoulder,
+            LeftTrigger,
+            RightTrigger,
+            X,
+            Y,
+            A,
+            B
+        };
+
+        enum Axis 
+        {
+            LeftStickX,
+            LeftStickY,
+            RightStickX,
+            RightStickY
+        };
+
+        public static FourCC kFormat = new FourCC('I', 'G', 'C', ' ');
+        public const int kMaxButtons = 14;
+        public const int kMaxAxis = 4;
+        
+        [InputControl(name = "dpad", template = "Dpad")]
+        [InputControl(name = "dpad/up", template = "Button", bit = (uint)Button.DpadUp)]
+        [InputControl(name = "dpad/right", template = "Button", bit = (uint)Button.DpadRight)]
+        [InputControl(name = "dpad/down", template = "Button", bit = (uint)Button.DpadDown)]
+        [InputControl(name = "dpad/left", template = "Button", bit = (uint)Button.DpadLeft)]  
+        [InputControl(name = "buttonSouth", template = "Button", bit = (uint)Button.A)]
+        [InputControl(name = "buttonWest", template = "Button", bit = (uint)Button.X)]
+        [InputControl(name = "buttonNorth", template = "Button", bit = (uint)Button.Y)]
+        [InputControl(name = "buttonEast", template = "Button", bit = (uint)Button.B)]
+        [InputControl(name = "leftStickPress", template = "Button", bit = (uint)Button.LeftStick)]
+        [InputControl(name = "rightStickPress", template = "Button", bit = (uint)Button.RightStick)]
+        [InputControl(name = "leftShoulder", template = "Button", bit = (uint)Button.LeftShoulder)]
+        [InputControl(name = "rightShoulder", template = "Button", bit = (uint)Button.RightShoulder)]
+        [InputControl(name = "leftTrigger", template = "Button", bit = (uint)Button.LeftTrigger)]
+        [InputControl(name = "rightTrigger", template = "Button", bit = (uint)Button.RightTrigger)]

```

Comment:
Triggers don't come through as axes on iOS?

Reply:
They don't, OS returns them as buttons. And buttons on iOS both hold bool and float value.

Reply:
I see. Why aren't we using the float value the triggers then?

Reply:
So on iOS triggers fall in same category as other buttons  - dpad, shoulder buttons, start, select. So it didn't make sense to take float for triggers and not take floats for other buttons.

I know that Gamepad class treat triggers as float input and those other buttons as bit values... that's why it's a bit confusing.

Maybe, let's make it like this - wire all those buttons (shoulder, dpad, select, start, triggers) to buttonValues variable as FLT format, as far as I understand InputSystem converts float to bool value if needed.

If user needs a direct OS bool value, he can call inputcontrol which will be wired to uint buttons? Not sure how to call such control, for leftShoulder wired to buttons, maybe should be called leftShoulderRaw or whatever.

What do you think?